ReshapeMetrics/OutputToElasticSearch.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using ReshapeMetrics.ElasticSearch;
+using ReshapeMetrics.Remote;
+
+namespace ReshapeMetrics
+{
+    public class OutputToElasticSearch : IOutputDescriptor
+    {
+        private readonly ElasticSearchUriTemplate serverElasticSearchUriTemplate;
+        private readonly ElasticSearchItemKeyGenerator keyGenerator;
+
+        public OutputToElasticSearch(string serverUri, string itemType = "metrics")
+        {
+            // Ensure that the generated URIs will have a trailing slash.
+            this.serverElasticSearchUriTemplate = new ElasticSearchUriTemplate(serverUri.TrimEnd('/') + '/');
+            keyGenerator = new ElasticSearchItemKeyGenerator(itemType);
+        }
+
+        public IOutput GetOutputFor(string relativePath, EnvironmentLookup environment)
+        {
+            var uri  = GenerateUri(environment);
+            var queue = GetOrCreateQueue(uri);
+            var key = keyGenerator.CreateFromTimeStamp(environment.Timestamp);
+            return new BufferedItem(queue, key);
+        }
+
+        public void Dispose()
+        {
+            var consumers = BeginShutdown();
+            WaitForShutdown(consumers).GetAwaiter().GetResult();

```

Comment:
`Wait()` does not unwrap `AggregateException`s. The objective here is to simulate the behaviour of an `await`, but synchronously.